include/swift/SIL/SILModule.h

```diff
@@ -178,6 +184,14 @@ class SILModule {
   // The list of SILCoverageMaps in the module.
   CoverageMapListType coverageMaps;
 
+  /// If non-null, the YAML file where remarks should be recorded.
+  std::unique_ptr<llvm::yaml::Output> OptRecordFile;
+
+  /// This is the underlying file descriptor of OptRecordFile.
+  ///
+  /// It is also owned by SILModule in order to keep their lifetime in sync.
+  std::unique_ptr<llvm::raw_fd_ostream> OptRecordFileDescriptor;
```

Comment:
I remember us having problems with wrapped streams and destructor order before, so please put this stream above the YAML stream in the field order. (Also, no reason to make it `raw_fd_ostream` specifically, right? May as well just say `raw_ostream` here.)

Reply:
No problem, thanks for making sure this feature was properly adapted to the Swift compiler!

Will make the above changes before re-test/merge.

Reply:
For the record, I've also renamed OptRecordFD to OptRecordRawStream and OptRecordFile to OptRecordStream to imply the types better.